---
permalink: /publications/
title: "Publications"
layout: single
author_profile: true
last_modified_at: 2024-02-01
---

[[Google Scholar](https://scholar.google.com/citations?hl=en&user=gzrxBacAAAAJ)] [[ORCID](https://orcid.org/0000-0002-7112-0224)] [[PubMed Bibliography](https://www.ncbi.nlm.nih.gov/myncbi/1d30fkpMdF_kft/bibliography/public/)]

*Asterisk (\*) denotes the Co-first Authors and Hashtag (#) denotes the Corresponding Authors.*

<!-- - title [[Web]()] [[Abstract](https://pubmed.ncbi.nlm.nih.gov/)] [[PDF](/assets/pdf/papers)] -->

- Chang, Z.\*, Xu, Y.\*, **Dong, X.\***, Gao, Y.# & Wang, C.# Single-cell and spatial multiomic inference of gene regulatory networks using SCRIPro. *Bioinformatics* 40, btae466 (2024). [[Web](https://academic.oup.com/bioinformatics/article/40/7/btae466/7716542?login=true)]

- Sun, F.\*, Li, H.\*, Sun, D.\*, Fu, S.\*, Gu, L.\*, Shao, X.\*, Wang, Q.\*, **Dong, X.\***, Duan, B.\*, Xing, F.\*, Wu, J.\*, Xiao, M.#, Zhao, F.#, Han, J.-D. J.#, Liu, Q.#, Fan, X.#, Li, C.#, Wang, C.# & Shi, T.# Single-cell omics: experimental workflow, data analyses and applications. *SCIENCE CHINA Life Sciences* (2024). [[Web](https://link.springer.com/article/10.1007/s11427-023-2561-0)]

- Ren, P.\*, Shi, X.\*, Yu, Z., **Dong, X.**, Ding, X., Wang, J., Sun, L., Yan, Y., Hu, J., Zhang, P., Chen, Q., Zhang, J., Li, T. & Wang, C.# Single-cell assignment using multiple-adversarial domain adaptation network with large-scale references. *Cell Reports Methods*, 3, 100577 (2023). [[Web](https://doi.org/10.1016/j.crmeth.2023.100577)] [[Abstract](https://pubmed.ncbi.nlm.nih.gov/37751689/)]

- Han, Y.\*, Wang, Y.\*, **Dong, X.\***, Sun, D., Liu, Z., Yue, J., Wang, H., Li, T.# & Wang, C.# TISCH2: expanded datasets and new tools for single-cell transcriptome analyses of the tumor microenvironment. *Nucleic Acids Research*, 51, D1029-D1037 (2022). [[Web](https://doi.org/10.1093/nar/gkac959)] [[Abstract](https://pubmed.ncbi.nlm.nih.gov/36321662/)] [[PDF](/assets/pdf/papers/Han%20et%20al_2022_TISCH2.pdf)]

- Shi, X.\*, Yu, Z.\*, Ren, P., **Dong, X.**, Ding, X., Song, J., Zhang, J., Li, T.# & Wang, C.# HUSCH: an integrated single-cell transcriptome atlas for human tissue gene expression visualization and analyses. *Nucleic Acids Research*, 51, D1425-D1431 (2022). [[Web](https://doi.org/10.1093/nar/gkac1001)] [[Abstract](https://pubmed.ncbi.nlm.nih.gov/36318258/)] [[PDF](/assets/pdf/papers/Shi%20et%20al_2022_HUSCH.pdf)]

- **Dong, X.\***, Tang, K.\*, Xu, Y., Wei, H., Han, T. & Wang, C. # Single-cell gene regulation network inference by large-scale data integration. *Nucleic Acids Research*, 50, e126 (2022). [[Web](https://academic.oup.com/nar/advance-article/doi/10.1093/nar/gkac819/6717821)] [[Abstract](https://pubmed.ncbi.nlm.nih.gov/36155797/)] [[PDF](/assets/pdf/papers/Dong%20et%20al_2022_Single-cell%20gene%20regulation%20network%20inference%20by%20large-scale%20data%20integration.pdf)]

- Xu, R.\*, Li, S.\*, Wu, Q.\*, Li, C.\*, Jiang, M.\*, Guo, L., Chen, M., Yang, L., **Dong, X.**, Wang, H., Wang, C.#, Liu, X.#, Ou, X.# & Gao, S.# Stage-specific H3K9me3 occupancy ensures retrotransposon silencing in human pre-implantation embryos. *Cell Stem Cell*, 29, 1051-1066.e8 (2022). [[Web](https://doi.org/10.1016/j.stem.2022.06.001)] [[Abstract](https://pubmed.ncbi.nlm.nih.gov/35803226/)]

- Sun, D.\*, Wang, J.\*, Han, Y.\*, **Dong, X.**, Ge, J., Zheng, R., Shi, X., Wang, B., Li, Z., Ren, P., Sun, L., Yan, Y., Zhang, P., Zhang, F.#, Li, T.# & Wang, C.# TISCH: a comprehensive web resource enabling interactive single-cell transcriptome visualization of tumor microenvironment. *Nucleic Acids Research*, 49, D1420-D1430 (2021). [[Web](https://academic.oup.com/nar/advance-article/doi/10.1093/nar/gkaa1020/5976978)] [[Abstract](https://pubmed.ncbi.nlm.nih.gov/33179754/)] [[PDF](/assets/pdf/papers/Sun%20et%20al_TISCH.pdf)]

- Zheng, R.\*, **Dong, X.\***, Wan, C., Shi, X., Zhang, X.# & Meyer, C.A.# Cistrome Data Browser and Toolkit: analyzing human and mouse genomic data using compendia of ChIP-seq and chromatin accessibility data. *Quantitative Biology*, 8, 267-276 (2020). [[Web](https://journal.hep.com.cn/qb/EN/10.1007/s40484-020-0204-7)][[PDF](/assets/pdf/papers/Zheng%20et%20al_2020_Cistrome%20Data%20Browser%20and%20Toolkit.pdf)]

- Chen, C.-H., Zheng, R., Tokheim, C., **Dong, X.**, Fan, J., Wan, C., Tang, Q., Brown, M., Liu, J.S., Meyer, C.A. # & Liu, X.S.# Determinants of transcription factor regulatory range. *Nature Communications*, 11, 2472 (2020). [[Web](https://www.nature.com/articles/s41467-020-16106-x)] [[Abstract](https://pubmed.ncbi.nlm.nih.gov/32424124/)] [[PDF](/assets/pdf/papers/Chen%20et%20al_2020_Determinants%20of%20transcription%20factor%20regulatory%20range.pdf)]

- Li, S.\*, Wan, C.\*, Zheng, R., Fan, J., **Dong, X.**, Meyer, C.A.# & Liu, X.S.# Cistrome-GO: a web server for functional enrichment analysis of transcription factor ChIP-seq peaks. *Nucleic Acids Research*, 47, W206â€“W211 (2019). [[Web](https://academic.oup.com/nar/article/47/W1/W206/5485528)][[Abstract](https://pubmed.ncbi.nlm.nih.gov/31053864/)][[PDF](/assets/pdf/papers/Li%20et%20al_2019_Cistrome-GO.pdf)]
